### PR TITLE
adding test for player module

### PR DESCRIPTION
**Pull request by shibani**
## Pivotal Link
https://www.pivotaltracker.com/story/show/160571496

## Description
- Create tests for Player methods
- Add Player methods

### CODE REVIEW

File: lib/game_modules/player.ex [hunk 1]
```diff
@@ -1,4 +1,14 @@
 defmodule Player do
 
+  def create_player(icon, type) do
+    %{icon: icon, type: type}
```
Comment (ecmendenhall) [inline lib/game_modules/player.ex]: Maps are a good start here. When I see a `type` field, it's a little bit of a code smell: it usually ends with a ["replace conditional with polymorphism"](https://refactoring.guru/replace-conditional-with-polymorphism) refactor.

Different types of players will behave differently but conform to the same interface. That's a great case for polymorphism. A simple map with a `type` attribute is a fine place to start, but hopefully you can lean on some of the tools Elixir provides to implement polymorphic players.

I'd start by reading up on [structs](https://elixir-lang.org/getting-started/structs.html), [protocols](https://elixir-lang.org/getting-started/protocols.html), and [polymorphism in Elixir](https://medium.com/elixirlabs/polymorphism-in-elixir-cd0c765b6929). Check out chapter 24 in the Elixir book too.

File: lib/game_modules/game.ex [hunk 2]
```diff
@@ -1,23 +1,50 @@
...
+      current_player == nil ->
+        List.first(player_list)
+      Map.equal?(current_player, List.last(player_list)) ->
+        List.first(player_list)
+      Map.equal?(current_player, List.first(player_list)) ->
+        List.last(player_list)
+    end
+  end
+
+  def square_to_rows_and_cols(square, row_size) do
```
Comment (shibani) [inline lib/game_modules/game.ex]: Leaving these helper methods here for now, to be moved to a better place later.

File: lib/game_modules/game.ex [hunk 3]
```diff
@@ -1,23 +1,50 @@
...
+    [
+      div(square - 1, row_size),
+      rem(square - 1, row_size)
+    ]
+  end
+
+  def rows_and_cols_to_square(row, col, row_size) do
```
Comment (shibani) [inline lib/game_modules/game.ex]: Leaving these helper methods here for now, to be moved to a better place later.

File: lib/game_modules/game.ex [hunk 4]
```diff
@@ -1,23 +1,50 @@
...
+    row * row_size + col + 1
+  end
+
+  def valid_move?(board, square, row_size) do
```
Comment (shibani) [inline lib/game_modules/game.ex]: Leaving this method here for now, to be moved to a better place later.

File: lib/game_modules/game.ex [hunk 5]
```diff
@@ -1,23 +1,50 @@
...
+        List.first(player_list)
+      Map.equal?(current_player, List.last(player_list)) ->
+        List.first(player_list)
+      Map.equal?(current_player, List.first(player_list)) ->
+        List.last(player_list)
+    end
+  end
+
+  def square_to_rows_and_cols(square, row_size) do
+    [
```
Comment (ecmendenhall) [inline lib/game_modules/game.ex]: Not a huge deal, but I think a tuple (or even a struct with `:row` and `:column` fields) would be a little more idiomatic here.

File: lib/game_modules/game.ex (hunk 4 above)
Comment (ecmendenhall) [inline lib/game_modules/game.ex]: Is this method as small and concise as it possibly could be, or can you extract and name another concept or two here?

File: lib/game_modules/game.ex [hunk 6]
```diff
@@ -1,23 +1,50 @@
...
+      rem(square - 1, row_size)
+    ]
+  end
+
+  def rows_and_cols_to_square(row, col, row_size) do
+    row * row_size + col + 1
+  end
+
+  def valid_move?(board, square, row_size) do
+    row_col_coords = Game.square_to_rows_and_cols(square, row_size)
```
[the remark this answers is not in the document]
Comment (shibani) [reply lib/game_modules/game.ex]: 💡

**Pull request merged**